### ISSUE TITLE
E2E tests - standardise input params, naming, use of logs etc.

### ISSUE THREAD

**Issue by govint** [vmware/docker-volume-vsphere#1297]
From the changes identified till date for #1286 below are items that need to be done to avoid continuous re-writes of the test code to use one approach over another.

1.  Convert all test groups to use gocheck vs. GO testing package - some of the tests in the repo are using gocheck while a couple are still using the earlier approach.
1. Each test group seems to be using different env. variables to get the VM IPs and ESX IPs to use for the test. Instead, resolve to using either the same env. vars in all test groups (easy to understand) or govc (as appropriate) via a set of common functions to get things like datastore names, VM names, ESX IP, what else thats usable for all test groups.
1. Fix the naming convention for all test functions - like Testgroupnamespecific-test-in-group()
1. Fix naming convention for variable names per test group.
1. Move as much of common functionality ((2) above) as functions into a util module that all test groups can leverage.
1. The volume properties test is a single function thats doing a bunch of tests, can this be broken down into separate tests - basically a test function verifies a single and specific functionality vs. multiple functions.

Of these, lets get the parameter generation addressed asap so all test groups have a standard way to use.

**Comment by shuklanirdesh82** [comment]
Thanks @govint here is the response from myside. Most of the items had been discussed before starting 0.15 sprint/release for automation effort. 

1. #1253 & #1096 are meant for 0.15 and already decided to convert them.
2. https://github.com/vmware/docker-volume-vsphere/blob/master/tests/utils/inputparams/testparams.go will be treated as a central place for common variable to consumed by testcases.
3/4/5: This issue is closed and decided to use common approach by taking reference of basic_test, restart_test, vmlisterner_test etc.
6. it has been discussed and will be taken care by #1096

 It is better to open a separate issue instead of super one. Please share your input if you think of anything missing here. 

I would suggest to close this issue and open a specific one or we may wait for planned tasks to be done for 0.15.

**Comment by govint** [comment]
@shuklanirdesh82 can you ensure that inputparams is the place to add all code for initializing common vars consumed by test cases.

Closing this issue as other issues are handled via other PRs/issues.
